src/main/scala/com/github/nscala_time/time/RichInterval.scala

```diff
@@ -21,16 +21,16 @@ class RichInterval(val underlying: Interval) extends AnyVal with PimpedType[Inte
     * }}}
     */
   def by(period: ReadablePeriod): List[DateTime] = {
-    val builder = List.newBuilder[DateTime]
-
-    var x = underlying.getStart
-
-    while (x <= underlying.getEnd) {
-      builder += x
-      x += period
+    @annotation.tailrec
+    def loop(current: DateTime, accumulator: List[DateTime]): List[DateTime] = {
+      if (current <= underlying.getEnd) {
+        loop(current + period, accumulator :+ current)
```

Comment:
`List#:+` is not efficient

Reply:
Thanks, replaced with `IndexedSeq#:+`.

Another option is to use prepend:
```scala
def by(period: ReadablePeriod): List[DateTime] = {
    @annotation.tailrec
    def loop(current: DateTime, accumulator: List[DateTime]): List[DateTime] = {
      if (current >= underlying.getStart) {
        loop(current - period, current +: accumulator)
      } else {
        accumulator
      }
    }

    loop(underlying.getEnd, Nil)
  }
```